.circleci/config.yml

```diff
@@ -66,23 +78,72 @@ jobs:
-          key: v2-node-{{ arch }}-{{ .Branch }}-{{ checksum "yarn.lock" }}-node-modules
+          name: Save root node_modules cache
+          key: v2-node-{{ arch }}-{{ .Branch }}-{{ checksum "yarn.lock" }}-root-node-modules
           paths:
             - node_modules
+      - save_cache:
+          name: Save DevTools node_modules cache
+          key: v2-node-{{ arch }}-{{ .Branch }}-{{ checksum "yarn.lock" }}-devtools-node-modules
+          paths:
+            # Packages manually listed because glob patterns are not supported;
+            # see # https://circleci.com/docs/2.0/configuration-reference/#save_cache
+            # Regenerate with `ls packages | grep 'devtools' | awk '{print "- packages/" $0 "/node_modules"}'`
+            - packages/react-devtools/node_modules
+            - packages/react-devtools-core/node_modules
+            - packages/react-devtools-extensions/node_modules
+            - packages/react-devtools-inline/node_modules
+            - packages/react-devtools-scheduling-profiler/node_modules
+            - packages/react-devtools-shared/node_modules
+            - packages/react-devtools-shell/node_modules
+      - save_cache:
+          name: Save other packages node_modules cache
+          key: v2-node-{{ arch }}-{{ .Branch }}-{{ checksum "yarn.lock" }}-package-node-modules
+          paths:
+            # Packages manually listed because glob patterns are not supported;
+            # see # https://circleci.com/docs/2.0/configuration-reference/#save_cache
+            # Regenerate with `ls packages | grep -v 'devtools' | awk '{print "- packages/" $0 "/node_modules"}'`
+            - packages/create-subscription/node_modules
+            - packages/dom-event-testing-library/node_modules
+            - packages/eslint-plugin-react-hooks/node_modules
+            - packages/jest-mock-scheduler/node_modules
+            - packages/jest-react/node_modules
+            - packages/react/node_modules
+            - packages/react-art/node_modules
+            - packages/react-cache/node_modules
+            - packages/react-client/node_modules
+            - packages/react-debug-tools/node_modules
+            - packages/react-dom/node_modules
+            - packages/react-fetch/node_modules
+            - packages/react-interactions/node_modules
+            - packages/react-is/node_modules
+            - packages/react-native-renderer/node_modules
+            - packages/react-noop-renderer/node_modules
+            - packages/react-reconciler/node_modules
+            - packages/react-refresh/node_modules
+            - packages/react-server/node_modules
+            - packages/react-test-renderer/node_modules
+            - packages/react-transport-dom-relay/node_modules
+            - packages/react-transport-dom-webpack/node_modules
+            - packages/scheduler/node_modules
+            - packages/shared/node_modules
+            - packages/use-subscription/node_modules
```

Comment:
This is something that will be easy to forget to update when we next add a new package. Will the (CI) failure behavior be obvious? (Will the error message reasonably direct folks to this bit of the config?)

Reply:
That sounds more robust.

Reply:
Can we run a yarn install for only the tests that can't use the global cache? Is that only the devtools test?

Reply:
Yeah we could do that. I'm just concerned that these missing folders may also cause problems in the future, so it would seem less risky to me if we just cached all node_modules folders. `react-native-renderer` and `react-transport-dom-relay` currently do have non-empty node_modules too (even though they only contain `scheduler`).

What do you think? I'll be happy to change this PR to either approach.

Reply:
Will defer to @rickhanlonii since he's more familiar with this than I am.

Reply:
> If we ever have 2+ nested deps that are different versions, this consolidated approach would cause problems.

Hmm, I was thinking of copying all the packages' node_modules folders into individual folders. So the cache will contain multiple `package_node_modules/<package name>/node_modules` folders alongside the root `node_modules` folder. This will allow us to restore all the packages' `node_modules` folders back to where `yarn install` created them. I think this should be pretty safe, if a little complicated.

My concern with running `yarn install` only for the new test is that other tests may suffer from silent problems if their conflicting deps aren't restored/installed. The set of conflicting deps may change over time as well.

Reply:
Sorry then. I misunderstood what you were suggesting. That sounds fine. 👍

Reply:
>  I think the consolidated cache idea seems problematic if we ever have conflicts between packages

Yarn workspaces hoist all of the dependencies in a package up to the root node_modules already unless there's a conflict. So to prevent conflicts, we could enforce all packages use the same version and add a test to compare the package.json files. What do you think about that?

Reply:
We already have several conflicts. Some of them probably arbitrary due to transient deps, others perhaps intentional (like `react-native-renderer` + `scheduler`).

Reply:
Btw this is what we do internally in xplat/js.